pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceProducerTest.java

```diff
@@ -161,6 +161,49 @@ public void testMsgKey() throws Exception {
         newConsumer2.close();
     }
 
+    @Test(timeOut = 20000)
+    public void testMsgKeyBatchingDisabled() throws Exception {
+        String argString = "%s -r 10 -u %s -m 500 -mk autoIncrement -db";
+        String topic = testTopic + UUID.randomUUID();
+        String args = String.format(argString, topic, pulsar.getBrokerServiceUrl());
+        Thread thread = new Thread(() -> {
+            try {
+                PerformanceProducer.main(args.split(" "));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+
+        Consumer<byte[]> newConsumer1 = pulsarClient.newConsumer().topic(topic).subscriptionName("sub-2")
```

Comment:
we don't have to publish or consume message. somehow, we need to assert that Producer is created with batching-disable option. you can also use reflection if any variable is private and you have to find the value of that field.